src/main/scala/com/advancedtelematic/campaigner/actor/CampaignScheduler.scala

```diff
@@ -68,9 +68,11 @@ class CampaignScheduler(director: DirectorClient,
       schedule(devices).pipeTo(self)
 
     case BatchToSchedule(devices) if devices.isEmpty =>
-      parent ! CampaignComplete(campaign.id)
-      // TODO: Should move to finished
-      context.stop(self)
+      campaigns.updateStatus(campaign.id).onComplete { _ =>
```

Comment:
Why not `forEach` if we don't handle errors here?

Reply:
To make sure that the actor stops no matter what happens in `updateStatus`

Reply:
Yeah that make sense to me 👍

Reply:
It's not safe to call `stop` on `onComplete`. The actor might be processing another message by the time `onComplete` is executed. See 2nd warning on this section of the akka docs: https://doc.akka.io/docs/akka/current/actors.html#ask-send-and-receive-future

Reply:
Thanks @simao, didn't know that. I'll update the code to avoid calling `stop` in the callback